Lib/http/client.py

```diff
@@ -944,21 +945,16 @@ def _tunnel(self):
         try:
             (version, code, message) = response._read_status()
 
+            self._proxy_response_headers = parse_headers(response.fp)
```

Comment:
This always does complicated parsing of the data despite these headers not being used in 99.999% of use cases.

It'd be better to just call _read_headers(response.fp) and store that raw data privately, and add a `get_proxy_response_headers()` public method that triggers the parsing into data structures.

The debuglevel > 0 print loop should just loop over the raw unparsed headers.